godot-core/src/export.rs

```diff
@@ -0,0 +1,161 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::builtin::meta::{ClassName, PropertyInfo};
+use crate::builtin::{GodotString, StringName};
+use crate::engine::global::{PropertyHint, PropertyUsageFlags};
+use crate::obj::GodotClass;
+
+use godot_ffi as sys;
+use sys::VariantType;
+
+/// Trait implemented for types that can be used as `#[export]` fields. This creates a copy of the
+/// value, for some type-specific definition of "copy". For example, `Array` and `Gd` are returned
+/// via `Share::share()` instead of copying the actual data.
+pub trait Export {
+    /// Creates a copy to be returned from a getter.
+    fn export(&self) -> Self;
+
+    /// The export info to use for an exported field of this type, if no other export info is specified.
+    fn default_export_info() -> ExportInfo;
+}
+
+/// Info needed for godot to understand how to export a type to the editor.
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct ExportInfo {
+    pub variant_type: VariantType,
+    pub hint: PropertyHint,
+    pub hint_string: GodotString,
+}
+
+impl ExportInfo {
+    /// Create a new `ExportInfo` with a property hint of
+    /// [`PROPERTY_HINT_NONE`](PropertyHint::PROPERTY_HINT_NONE).
+    pub fn new_none(variant_type: VariantType) -> Self {
+        Self {
+            variant_type,
+            hint: PropertyHint::PROPERTY_HINT_NONE,
+            hint_string: GodotString::new(),
+        }
+    }
```

Comment:
Maybe name it `with_hint_none()` or so?
Otherwise it's not really clear that "none" refers to the property hint.